### PR TITLE
Few random things

### LINKED ISSUE #3: Use XDG_RUNTIME_DIR
[This article](https://medium.com/@jdxcode/12-factor-cli-apps-dd3c227a0e46) makes a great point about following XDG, and we totally should! We were torn about the directory earlier, and this should be a simple enough change!

Edit: It appears that mac does not have this!

### PR DESCRIPTION
**Pull request by FatBoyXPC**
Implement tests, XDG spec, replace pstree with psutil, and prevent `shtuff` from blowing up when called without any args or a bad target.

This fixes #3, #4, and #5.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,10 +30,21 @@ running tail.
...
 ```
 
+Run tests:
+
+```bash
+$ rm -rf tests/build && pip install -t tests/build .
+$ python -m unittest
+```
+
+If weird things happen, it might be worth running `find . -name __pycache__ | xargs rm -rf`
```
Comment (jfly) [inline README.md]: Did this ever happen again? I'd vote for either elaborating on "weird stuff" or getting rid of the comment. I honestly don't even remember what happened anymore.

File: shtuff.py [hunk 2]
```diff
@@ -111,11 +112,17 @@ def get_cmd_file(pid):
...
 def find_nearest_shtuff_process():
-    return subprocess.run(
-        """pstree -lps $$ | grep -Eo "shtuff\([0-9]+\)" | grep -Eo '[0-9]+'""",
-        shell=True,
-        capture_output=True,
-    ).stdout.decode().strip()
+    def ppid(process):
+        parent = process.parent()
+        if parent is None:
+            return
```
Comment (jfly) [inline shtuff.py]: This works, buts "more pythonic" to explicitly return None instead. Plain return is just for functions that don't return anything (as in, people should never use their return value)

**Review by jfly — APPROVED** [review]
Damn fine work! Loving the test infra.

File: README.md (hunk 1 above)
Comment (FatBoyXPC) [reply README.md]: It didn't, and I don't remember exactly _what_ happened. I believe after we did this that we also ran `pip install -t tests/build .`.

File: setup.cfg [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[install]
```
Comment (jfly) [inline setup.cfg]: Nit: Maybe add a comment here linking back to the thread where you learned about this trick?

**Pull request merged**